Snowflake.Data/Core/SFSessionProperty.cs

```diff
@@ -140,6 +171,26 @@ private static bool IsRequired(SFSessionProperty sessionProperty, SFSessionPrope
                 return sessionProperty.GetAttribute<SFSessionPropertyAttr>().required;
             }
         }
+
+        private static string ParseAccount(string rawAccount)
+        {
+            // filter the .<region> or .global
+            string parsedAccount = rawAccount;
+            int firstDot = parsedAccount.IndexOf('.');
+            if (firstDot != -1)
+            {
+                parsedAccount = parsedAccount.Substring(0, firstDot);
+            }
+
+            // filter the connection group
+            int firstDash = parsedAccount.IndexOf('-');

```

Comment:
Today this is valid only of `global` is included in the subdomain?

Reply:
Yes, the filtering of the connection group should only be done if it contains `global`. I would also use `LastIndexOf` instead of `IndexOf`. Today we don't allow dashes in the account name, but if we did in the future, this may parse out too much information. The external ID will always appended to the end so `LastIndexOf` would always work.

Reply:
https://github.com/snowflakedb/snowflake-connector-net/tree/prep-1.0.12#create-a-connection

Reply:
We suggested users don't use account but rather using hostname.